python/pyarrow/tests/test_extension_type.py

```diff
@@ -276,8 +276,7 @@ def test_ext_type_byte_width():
 
     # Test for non fixed-size binary types
     ty = LabelType()
-    with pytest.raises(ValueError, match="Non-fixed width type"):
-        _ = ty.byte_width
+    assert ty.byte_width == -1
```

Comment:
I think we want that the extension type is consistent here with the normal data types, which raise an error in this case (you can check how the error is raised in the base DataType class)

Reply:
You mean that the extension type should make erorr? In this case, the extension type return -1.